### PR TITLE
Add tool battery mount for vehicles (reverse battery compartment mod).

### PR DESCRIPTION
**Pull request by Tamiore**
#### Summary
Content "Add tool battery mount for vehicles (reverse battery compartment mod)."

#### Purpose of change

Battery compartment mod allows you to use vehicle batteries in handheld tools. 
This PR adds a vehicle part — "tool battery mount" — that allows you to use conventional batteries from handheld tools as primary battery in a vehicle. Basically the reverse functionally of  battery compartment mod.

#### Describe the solution
Same as normal tools require a mod (battery compartment mod) to use vehicle batteries, vehicles now require a mod (vehicle part that works similar to `storage battery case`) to use conventional batteries from handheld tools. 
Only accepts a single conventional battery per vehicle tile, takes up fuel source slot for that tile, the battery itself is quite vulnerable when installed like that.

#### Testing

Mostly .json change, works fine locally.
There is a single line in vehicle.cpp, but I don't think any extensive testing is required for that one one.

#### Additional context
Speaking of the line in vehicle.cpp: is it actually required?
I feel like everything that line does (and similar line for  `storage battery case`) has already been jsonized.

Also, implementation has been somewhat discussed here: https://discourse.cataclysmdda.org/t/best-way-to-implement-reverse-battery-compartment-mod-within-current-code-structure/26718

### CODE REVIEW

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/best-way-to-implement-reverse-battery-compartment-mod-within-current-code-structure/26718/27

**Comment by Tamiore** [comment]
@kevingranade 
Anything needs to be done before this can be merged?

File: data/json/vehicleparts/vp_flags.json [hunk 1]
```diff
@@ -236,6 +236,12 @@
     "context": [ "vehicle_part" ],
     "requires_flag": "BATTERY_MOUNT"
   },
+  {
+    "id": "NEEDS_HANDHELD_BATTERY_MOUNT",
+    "type": "json_flag",
+    "context": [ "vehicle_part" ],
```
Comment (kevingranade) [inline data/json/vehicleparts/vp_flags.json:242]:
```suggestion
```

File: data/json/vehicleparts/vp_flags.json [hunk 2]
```diff
@@ -301,6 +307,11 @@
     "type": "json_flag",
     "context": [ "vehicle_part" ]
   },
+  {
+    "id": "HANDHELD_BATTERY_MOUNT",
+    "type": "json_flag",
+    "context": [ "vehicle_part" ]
```
Comment (kevingranade) [inline data/json/vehicleparts/vp_flags.json:313]:
```suggestion
    "type": "json_flag"
```

**Comment by kevingranade** [comment]
No blockers, it just got lost in the shuffle a bit and it took a while for me to get back around to it.
Did some light testing in game, it works great, just need to wait on a test rerun after I updated the json a bit (not your fault, it changed since you last updated it).

**Pull request merged**